crates/swc_css_parser/src/lexer/mod.rs

```diff
@@ -1007,19 +996,19 @@ where
         maybe_third: Option<char>,
     ) -> LexResult<bool> {
         // Look at the first code point:
-        let first = maybe_first.or(self.input.cur());
+        let first = maybe_first.or(self.cur());
 
         match first {
             // U+002B PLUS SIGN (+)
             // U+002D HYPHEN-MINUS (-)
             Some('+') | Some('-') => {
-                match maybe_second.or(self.input.peek()) {
+                match maybe_second.or(self.next()) {
                     // If the second code point is a digit, return true.
                     Some(second) if second.is_digit(10) => return Ok(true),
                     // Otherwise, if the second code point is a U+002E FULL STOP (.) and the
                     // third code point is a digit, return true.
                     Some('.') => {
-                        if let Some(third) = maybe_third.or(self.input.peek_ahead()) {
+                        if let Some(third) = maybe_third.or(self.next_next()) {
```

Comment:
Seems like an inefficeint utf8 decoding.

Reply:
@kdy1 Can you clarify about `inefficeint utf8 decoding`?

Reply:
Do you mean use something like:
```rust
match self.cur() {
  Some(ch) => ch == c as char,
  _ => false,
}
```

If yes we can rewrite on it, but before we don't use, maybe we can postpone it to the next PR, it is easy, but I wan't to mix it